### ISSUE TITLE
Custom dataset

### ISSUE THREAD

**Issue by ZeynepP** [liwk19/NORA#1]
Hi,

Congratulations on your paper, and thank you for sharing your code! I'm interested in applying your approach to a custom dataset. Could you please provide some guidance or steps on how to adapt the existing code to work with a different dataset? Any tips or advice would be greatly appreciated.

Thanks in advance for your help!

Best regards,
ZP

**Comment by weikai-li** [comment]
Hi ZeynepP,

Thank you for being interested in our work! We hope our codes can be helpful to the community, so we would like to first mention whether NORA is suitable for your situation. If you just want to calculate the influence of node removal on GNN's prediction, and if your dataset is not very big, then you can simply take turns removing each node from the original graph, apply the GNN model, and record the output changes. This way you can get the ground truth results within a few hours or a few days. As a reference, on the ogbn-arxiv dataset, we can get the ground truth results within 41 hours for five models with five different random seeds. It is much longer than using an approximation method like NORA, but I think it is still acceptable in many situations. If your dataset is too large to run the ground truth results, or if you want to use NORA as a baseline approximation method, then adapting NORA's codes is suitable for you. Besides, NORA is mainly designed for common GNN models that are based on 1-hop message passing. It might not perform well on some new GNN architectures (but feel free to use it as a baseline).

Probably the most convenient way is to adapt some of our codes to your repository. Here are a few steps: (1) You can write your codes for training your GNN model on your dataset as normal. (2) Run the ground truth results for at least 10% of nodes as the validation set to tune the hyper-parameters. (3) Edit the forward function in your GNN model to allow it to return a list of hidden representations of each layer (similar to the "return_hidden" option in /planetoid/models.py). (4) Run NORA to approximate the node influence (please refer to the "nora" function in main.py). (5) Use the "performance" function in evaluate.py to evaluate NORA's performance on the validation set and tune the hyper-parameters. The hyper-parameters are --self_buff, --grad_norm, --k1, --k2, and --k3 in main.py.

I hope my reply could be helpful. Please feel free to contact me if you have further questions.

Best,
Weikai Li

**Comment by ZeynepP** [comment]
Thanks for the details. I will be using a custom twitter dataset which is bigger than ogbn-arxiv dataset. 
Bests
ZP

**Comment by weikai-li** [comment]
I am glad I was able to help. Please feel free to contact me if you have further questions.
